compiler/qsc_frontend/src/compile.rs

```diff
@@ -59,6 +59,8 @@ pub struct AstPackage {
 #[derive(Debug, Default)]
 pub struct SourceMap {
     sources: Vec<Source>,
+    /// if the sources are from a project, then this is the project root directory.
+    project_root_dir: Option<Arc<str>>,
```

Comment:
So, based on all the things we've discussed, I suggest naming this something other than "project root" to reduce confusion. It's not always the "project root" in the sense that most people would understand it (i.e. wherever the qsharp.json is).

What about one of:

- `common_prefix`
- `common_path` (inspired by Python having a `commonpath()` function
- `common_root_dir`
- `common_root`


And I'd clarify in the comments that this value doesn't always have to be equal to the project root.